### PR TITLE
Comprehensive SVML unit test

### PR DESCRIPTION
**Pull request by anton-malakhov**
Unit test covering code-generation for more SVML functions at once. It creates foundation for future extensions covering more functions or documenting these, which will not work in Numba.

Currently, it covers 12 numpy or 20 math functions in 48 modes: (2 types, 3 arch x 2 flags x 4 use-cases). The list of SVML functions not yet supported by the test or not represented in Python include: pow atan2 cbrt cdfnorm cdfnorminv ceil cosd erfcinv erfinv exp10 exp2 floor fmod hypot invsqrt log2 logb nearbyint rint round sind trunc.

### CODE REVIEW

**Comment by anton-malakhov** [comment]
@stuartarchibald could you please quickly check if this code is compliant to any python/numba/style practices/guidelines I might have missed here. E.g. I'm concerned with lots of TODO marks I left there. Thanks!

File: numba/tests/test_svml.py [hunk 1]
```diff
@@ -6,15 +6,172 @@
...
+            fn, contains, avoids = combo_svml_usecase(*args)
+            # look for specific patters in the asm for a given target
+            with override_env_config('NUMBA_CPU_NAME', vlen2cpu[vlen]), \
+                 override_env_config('NUMBA_CPU_FEATURES', ''):
+                # recompile for overridden CPU
+                jit = compile_isolated(fn, (numba.int64,), flags=flags)
+            asm = jit.library.get_asm_str()
+            missed = [pattern for pattern in contains if not pattern in asm]
+            found  = [pattern for pattern in avoids if pattern in asm]
+            assert not missed and not found, "While expecting %s,\n\tit contains %s:\n%s\nwhen compiling %s" % \
```
Comment (stuartarchibald) [inline numba/tests/test_svml.py]: Perhaps use members of the `self.assert*` family to be consistent with other test code.

File: numba/tests/test_svml.py [hunk 2]
```diff
@@ -6,15 +6,172 @@
...
+                (str(missed), str(found), '\n'.join(
+                [line for line in asm.split('\n') if re.search('\$[a-z_]\w+,', line)]), fn.__doc__)
```
Comment (stuartarchibald) [inline numba/tests/test_svml.py]: Perhaps hoist the regex to class member level?

**Comment by stuartarchibald** [comment]
@anton-malakhov looks good. Style wise, Numba tries to format as PEP8, `autopep8` can do this for you if desirable. I added a couple of comments, but overall the approach looks good. In relation to `TODOs` they are useful left in, perhaps just add a new issue for code that isn't written if it isn't going to make it into this PR? The only other comment I have is to please include plenty of docstrings/comments over what is going on and how to extend the testing in the future. I'll do a full review when you remove the `[WIP]` from the title :) Thanks for the effort!

**Comment by anton-malakhov** [comment]
Thank you, @stuartarchibald ! I've addressed your comment and aligned the code with autopep8 suggestions (except few cases of excessive indentation). I also extended it onto `sqrt` function and `float32` type. I think it's good enough for now and the rest of the functions and modes can be covered when we'll have additional motivation/interest for testing these.

**Comment by stuartarchibald** [comment]
Thanks for the fixes @anton-malakhov , looks good. Merging.

**Pull request merged**